starknet_py/net/udc_deployer/deployer.py

```diff
@@ -0,0 +1,133 @@
+from starknet_py.utils.contructor_args_translator import translate_constructor_args
+from starknet_py.utils.crypto.facade import pedersen_hash
+from starknet_py.utils.data_transformer.universal_deployer_serializer import (
+    universal_deployer_serializer,
+    deploy_contract_abi,
+)
+from starknet_py.utils.sync import add_sync_methods
+
+
+ContractDeployment = NamedTuple("ContractDeployment", [("udc", Call), ("address", int)])
+
+
+@add_sync_methods
+class Deployer:
+    """
+    Deployer used to deploy contracts through Universal Deployer Contract (UDC)
+    """
+
+    def __init__(
+        self,
+        account: AccountClient,
+        *,
+        deployer_address: AddressRepresentation = DEFAULT_DEPLOYER_ADDRESS,
+        unique: bool = True,
+    ):
+        """
+        :param account: AccountClient used to sign and send transactions
+        :param deployer_address: Address of the UDC. Is set to the address of
+            the default UDC (same address on real nets and devnet) by default.
+            Must be set when using custom network other than devnet.
+        :param unique: Boolean determining if the salt should be connected with the account's address. Default to True
+        """
+
+        self.account = account
+        self.deployer_address = parse_address(deployer_address)
+        self.unique = unique
+
+    async def prepare_contract_deployment(
+        self,
+        *,
+        class_hash: Hash,
+        salt: Optional[int] = None,
+        abi: Optional[List] = None,
+        calldata: Optional[Union[List, dict]] = None,
+        max_fee: Optional[int] = None,
+        auto_estimate: bool = False,
+    ) -> InvokeFunction:
+        """
+        Prepares deploy invoke transaction
+
+        :param class_hash: The class_hash of the contract to be deployed
+        :param salt: The salt for a contract to be deployed. Random value is selected if it is not provided
+        :param abi: ABI of the contract to be deployed
+        :param calldata: Constructor args of the contract to be deployed
+        :param max_fee: Max amount of Wei to be paid when executing transaction
+        :param auto_estimate: Use automatic fee estimation, not recommend as it may lead to high costs
+        :return: NamedTuple with invoke_tx and address fields
+        """
+        call, address = await self.create_deployment_call(
+            class_hash=class_hash, salt=salt, abi=abi, calldata=calldata
+        )
+
+        transaction = await self.account.sign_invoke_transaction(
+            calls=call, max_fee=max_fee, auto_estimate=auto_estimate
+        )
+
+        return namedtuple(
+            typename="InvokeAndAddress", field_names=["invoke_tx", "address"]
+        )(transaction, address)
+
+    async def create_deployment_call(
+        self,
+        *,
+        class_hash: Hash,
+        salt: Optional[int] = None,
+        abi: Optional[List] = None,
+        calldata: Optional[Union[List, dict]] = None,
```

Comment:
```suggestion
        self,
        class_hash: Hash,
        *,
        salt: Optional[int] = None,
        abi: Optional[List] = None,
        calldata: Optional[Union[List, dict]] = None,
```

I think we can still allow passing class_hash as a positional argument as it is always required: I think it's more usable that way.

Reply:
I see your point, but I am not sure if it is worth it. My fear is that they might change class_hash to something else that we cannot predict and it is much easier to deprecate the old one with keyword arguments. We had a situation with transaction parameter somewhere that we couldn't imagine being changed and yet it happened. On the other hand I don't know what else could be used here in the future 🤔 
@war-in what do you think?

Reply:
I understand the issue with deprecation and it makes sense: however how likely is that the class hash will change to something else at this point?

Reply:
Changing class_hash to something else is quite a big issue. I think we can change it to the positional parameter

Reply:
I am fine with that